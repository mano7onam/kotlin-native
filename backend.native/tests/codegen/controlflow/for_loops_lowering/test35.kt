package codegen.controlflow.for_loops_lowering.test35

import kotlin.test.*

@Test fun runTest() {
  val arr = arrayOf(4, 5, 6, 14, 15, 16, 24, 25, 26, 34, 35, 36, 0, 50, 60, 70)
  fun foo() : Int = 10
  fun bar() : Int { println("bar"); return 14 }
  fun tre() : Int { println("tre"); return 2 }
  for (i in arr) {
    val arr1 = arrayOf(14, 24, 24)
    if (i in arr1) {
      println(i)
    }
    val rng1 = 0 .. foo()
    if (i in rng1) {
      println(i)
    }
    val rng2 = 0 until bar()
    if (i in rng2) {
      println(i)
    } 
    val rng3 = 15 downTo 0
    if (i in rng3 step foo()) {
      println(i)
    }
    val inds1 = arrayOf(1, 2, 3, 4, 5, 6, 7, 8, 9, 10).indices
    if (i in inds1) {
      println(i)
    }
    if (i in inds1 step tre()) {
      println(i)
    }
  }
}
