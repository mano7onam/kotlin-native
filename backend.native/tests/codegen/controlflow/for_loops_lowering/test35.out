4
bar
4
4
tre
4
5
bar
5
5
5
tre
6
bar
6
6
tre
6
14
bar
tre
bar
15
tre
bar
tre
24
bar
tre
bar
tre
bar
tre
bar
tre
bar
tre
bar
tre
0
bar
0
0
tre
0
bar
tre
bar
tre
bar
tre
